clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
I don't think that choosing JSON as our API prevents users from using whatever format they want externally; it just means that they need to convert it to the form in which we expect the data before they pass it to us.  That's true no matter what form we choose for the API -- whether it be JSON or a custom struct -- so IMHO, this is not a good argument against JSON.

BTW, another reason for using JSON instead of a custom struct is that JSON allows users to add their own per-method parameters, which they might use in their own interceptor implementations.  A struct is not extensible the way that JSON is.  In the C-core implementation, we allow individual filters to parse the JSON structure for the parameters that they're interested in, which provides a lot of extensibility.

Ultimately, I think that whatever we choose, we are making a decision for the users.  While it's certainly true that not everyone will be happy with whatever we choose, I think we should still make the choice that we think will work best overall.  I think it's important to provide the same format across all gRPC language implementations, and I think JSON is the best choice for that format.

I understand that this adds a dependency, and I understand that that's not ideal.  But I think it's worth adding a build-time dependency in order to support having the same format in all languages.

As I said before, if you want to discuss this further, please throw something on my calendar.  Thanks!